src/main/java/com/ibm/stocator/fs/swift/auth/AuthRequest.java

```diff
@@ -0,0 +1,64 @@
+package com.ibm.stocator.fs.swift.auth;
+
+import java.io.UnsupportedEncodingException;
+
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+
+import org.codehaus.jettison.json.JSONArray;
+import org.codehaus.jettison.json.JSONException;
+import org.codehaus.jettison.json.JSONObject;
+
+public class AuthRequest extends HttpPost {
+  private AccountConfiguration config;
+
+  public AuthRequest(AccountConfiguration conf) {
+    super(conf.getAuthUrl());
+    config = conf;
+    String authMethod = config.getAuthMethod();
+    String messageBody = "";
+
+    messageBody = createKeystoneV3Json();
+
+    // TODO(djalova) authenticate for keystone v1
```

Comment:
you mean Keystone V2...this is POST
but swiftaith (V1 ) this is GET.
Not sure how it fits with this class that extends HttpPOST

It wold be perfect to have something generic..like some class that will provide auth methods and internally will call KeystoneV2implementation, KeystoneV3impl, SwiftauthImpl, etc...

This way it will be more generic and we can add other authentications methods in the future